Es external index

LinhongLiu opened the pull request:
## What changes were proposed in this pull request?

(Please fill in changes proposed in this fix)


## How was this patch tested?

(Please explain how this patch was tested. E.g. unit tests, integration tests, manual tests)


(If this patch involves UI changes, please attach a screenshot; otherwise, remove this)

LinhongLiu committed:
add es dependency

LinhongLiu committed:
OapTransportClient

LinhongLiu committed:
es module interface

LinhongLiu committed:
es module with create/insert/drop

LuciferYang commented on src/main/java/org/apache/spark/sql/execution/datasources/parquet/OapTransportClient.java:
```diff
@@ -0,0 +1,94 @@
+package org.apache.spark.sql.execution.datasources.parquet;
```
Add Spark LICENSE?

LuciferYang commented on src/main/scala/org/apache/spark/sql/execution/datasources/oap/index/OapEsIndex.scala:
```diff
@@ -0,0 +1,255 @@
+
+  private val client = getEsClient
+
+  private val mappings = {
```
`number_of_shards` and `number_of_replicas` should be configurable

LuciferYang commented on src/main/scala/org/apache/spark/sql/execution/datasources/oap/index/OapEsIndex.scala:
```diff
@@ -0,0 +1,255 @@
+      bulkRequest.add(indexRequest)
+    }
+    bulkRequest.execute().actionGet()
+    client.admin().indices().refresh(new RefreshRequest(tableName)).actionGet()
```
We need to make sure whether `refresh` operation is necessary to let indexes to be available.

LinhongLiu replied:
yes, it's necessary.

LuciferYang commented on src/main/scala/org/apache/spark/sql/execution/datasources/oap/index/OapEsIndex.scala:
```diff
@@ -0,0 +1,255 @@
+   */
+  override protected def getEsClient: TransportClient = {
+    val settings = Settings.builder
+      .put("cluster.name", "elasticsearch")
```
cluster.name should be configurable, not all cluster is called `elasticsearch`

LinhongLiu replied:
let's do this after all configs figured out. for example: host, port.

LuciferYang replied:
ok

LinhongLiu committed:
fix something

LinhongLiu committed:
disable test suite

Commit:
Refactor FiberSensor (#823)

## What changes were proposed in this pull request?

1st subtask of steps commented in #757. 

- Refactored the class hierarchy of `FiberSensor`
- Combine 2 `FiberxxSensor` to 1
- Remove unused things

## How was this patch tested?

Existing tests.

carsonwang committed:
Spark 2.2 support - update code for spark 2.2   (#813)

* add profile for spark 2.1 and 2.2

* Move spark code to spark2.1

* add 2.2 SqlBase.g4

* add 2.2 ColumnBatch.java

* add 2.2 DAGScheduler.scala

* Add 2.2 DataSource.scala

* add 2.2 DataSourceStrategy

* revert changes in FileFormatWriter when moving

* add 2.2 FileFormatWriter.scala

* add 2.2 FileSourceStrategy.scala

* revert changes in InsertIntoHadoopFsRelationCommand when moving code

* add 2.2 OutputWriter.scala

* fix FileFormatWriter

* add 2.2 DataSourceScanExec

* update 2.2 SparkSqlParser

* update HiveThriftServer2.scala

* update 2.2 SparkSQLCLIDriver

* update 2.2 ApiRootResource.scala

* update 2.2 BlockId

* update 2.2 BitSet

* fix OapConf

* update travis to test 2.2

* revert changes in 2.1 OutputWriter

* Fix CaseINsensitiveMap compatitible issue

* fix InputFileNameHolder compatitible issue

* fix compatitible issue in indexPlans

* Fix ParquetFilters inaccessible issue

* Fix OapIndexInfoStatusSerDe compatible issue

* Fix FileIndex compatible issue

* Add LogicalPlanAdapter

* Add AggregateFunctionAdapter

* Fix listFiles

* Add FileSourceScanExecAdapter

* Fix OapConf

* Add RpcEndpointRefAdapter

* Fix ApiRootResource

* Add OapSessionStateBuilder and fix OapEnv and OapSession for 2.2

* Fix OutputWriter

* Fix more

* sqlConf -> sqlContext.conf

* Fix listFiles again

* Fix OapStrategies

* change strategies order

* update java version and fix SparkConf issue in test

* Fix task write result in FileFormatWriter

* add PARQUET_INT64_AS_TIMESTAMP_MILLIS as LuciferYang suggests

* import ColumnarBatchScan

* update ColumnarBatchScan

* fix unit tests

* update trvis

* update travis to test both 2.1 and 2.2

* remove FileIndex.scala

* update OapEnv and readme

* tweak

LuciferYang committed:
Import Vectorized Reader code from spark (#836)

## What changes were proposed in this pull request?
SubTask 1 of #825(#829),  import `VectorizedRleValuesReader`, `VectorizedPlainValuesReader` ,`VectorizedRleValuesReader` from Spark Code include 2.1 & 2.2, will change some access modifer of these class.

## How was this patch tested?

mvn test pass

zhixingheyi-tian committed:
Spark 2.3 support -  update pom.xml for Spark 2.3 (#840)

* update pom.xml

* Spark 2.3 support - import spark 2.2 source code

* Revert "update pom.xml"

This reverts commit 24e4b0cd740d50071a9255b0782b9457b44330dd.

* quit Revert "Revert "update pom.xml""

This reverts commit 8b5914bace850ba9fdd4e0509b1f1cfe131e1ce5.

* Revert "Spark 2.3 support - import spark 2.2 source code"

This reverts commit 7d7052944e0aa162e00be7c7d96e93149b06b80a.

* update pom.xml with exclude

LinhongLiu committed:
add DDL to create/drop ES index

LinhongLiu committed:
add spark 2.2 support

LinhongLiu committed:
integrate code
